### PR TITLE
Update instruction data to compatible with Anchor

### PR DESCRIPTION
**Pull request by seanyoung**
See https://github.com/hyperledger/solang/pull/1105

### CODE REVIEW

File: src/contract.ts [hunk 1]
```diff
@@ -206,24 +201,10 @@ export class Contract {
...
-            // lamports to send to payable constructor
-            encodeU64(BigInt(value)),
-            // hash of contract name
-            Buffer.from(hash.substr(2, 8), 'hex'),
-            // PDA seeds
-            encodeSeeds(seeds),
-            // abi encoded constructor arguments
-            input,
-        ]);
+        const discriminator_image = 'global:' + snakeCase('new');
```
Comment (LucasSte) [inline src/contract.ts:204]:
```suggestion
        const discriminator_image = 'global:new';
```

File: yarn.lock [hunk 2]
```diff
@@ -273,2299 +273,2305 @@
...
+  "resolved" "https://registry.npmjs.org/chokidar/-/chokidar-3.5.2.tgz"
+  "version" "3.5.2"
+  dependencies:
+    "anymatch" "~3.1.2"
+    "braces" "~3.0.2"
+    "glob-parent" "~5.1.2"
+    "is-binary-path" "~2.1.0"
+    "is-glob" "~4.0.1"
+    "normalize-path" "~3.0.0"
+    "readdirp" "~3.6.0"
```
Comment (LucasSte) [inline yarn.lock:834]: There is no need to change this file.

Comment (LucasSte) [reply yarn.lock:834]: You should run `yarn --freeze-lockfile`, if you are not adding a new dependency to the repository.

**Review by LucasSte — CHANGES_REQUESTED** [review]
Did you run `yarn fmt` and `yarn lint`?

**Comment by seanyoung** [comment]
> Did you run `yarn fmt` and `yarn lint`?

Yes, those should be checked in CI now

**Review by LucasSte — APPROVED** [review]

**Pull request merged**